gitlab.go

```diff
@@ -595,6 +598,14 @@ func (c *Client) NewRequest(method, path string, opt interface{}, options []Requ
 		return nil, err
 	}
 
+	for _, fn := range c.requestOptions {
+		if fn == nil {
+			continue
+		}
+		if err := fn(req); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Instead  of adding a new block, I prefer to concat the slices:

```go
for _, fn := range append(c.defaultRequestOptions, options...) {
		if fn == nil {
			continue
		}
		if err := fn(req); err != nil {
			return nil, err
		}
	}
```

Reply:
I would have expected that you'd prefer avoiding the (unnecessary) allocation, but done as requested.

Reply:
Fair point... Yet I would hope (and expect a little bit) that the compiler is smart enough to compile this down to effective bytecode 😊